packages/app/src/store/studio-store.ts

```diff
@@ -0,0 +1,907 @@
+      if (this.isFailed || !this._trustEvent(event)) return
+
+      const $el = window.UnifiedRunner.CypressJQuery(event.target)
+
+      if (this._isAssertionsMenu($el)) {
+        return
+      }
+
+      this._closeAssertionsMenu()
+
+      if (!this._shouldRecordEvent(event, $el)) {
+        return
+      }
+
+      const name = this._getName(event, $el)
+      const message = this._getMessage(event, $el)
+
+      if (name === 'change') {
+        return
+      }
+
+      let selector: string | undefined = ''
+
+      if (name === 'click' && this._matchPreviousMouseEvent($el)) {
+        selector = this._previousMouseEvent?.selector
+      } else {
+        selector = getCypress().SelectorPlayground.getSelector($el)
+      }
+
+      this._clearPreviousMouseEvent()
+
+      if (name === 'type' && !message) {
+        return this._removeLastLogIfType(selector)
+      }
+
+      const updateOnly = this._updateLastLog(selector, name, message)
+
+      if (updateOnly) {
+        return
+      }
+
+      if (name === 'type') {
+        this._addClearLog(selector)
+      }
+
+      this._addLog({
+        selector,
+        name,
+        message,
+      })
+    },
+
+    _removeLastLogIfType (selector?: string) {
+      const lastLog = this.logs[this.logs.length - 1]
+
+      if (lastLog.selector === selector && lastLog.name === 'type') {
+        return this.removeLog(lastLog.id)
+      }
+    },
+
+    removeLog (commandId?: number) {
+      const index = this.logs.findIndex((command) => command.id === commandId)
+      const log = this.logs[index]
+
+      this.logs.splice(index, 1)
+
+      this._generateBothLogs(log).forEach((commandLog) => {
+        getEventManager().emit('reporter:log:remove', commandLog)
+      })
+    },
+
+    _addLog (log: StudioLog) {
+      log.id = this._getId()
+
+      this.logs.push(log)
+
+      this._generateBothLogs(log).forEach((commandLog) => {
+        getEventManager().emit('reporter:log:add', commandLog)
+      })
+    },
+
+    _addAssertion ($el: HTMLElement, ...args: unknown[]) {
+      const id = this._getId()
+      const selector = getCypress().SelectorPlayground.getSelector($el)
+
+      const log: StudioLog = {
+        id,
+        selector,
+        name: 'should',
+        message: args,
+        isAssertion: true,
+      }
+
+      this.logs.push(log)
+
+      const reporterLog = {
+        id,
+        selector,
+        name: 'assert',
+        message: this._generateAssertionMessage($el, args),
+      }
+
+      this._generateBothLogs(reporterLog).forEach((commandLog) => {
+        getEventManager().emit('reporter:log:add', commandLog)
+      })
+
+      this._closeAssertionsMenu()
+    },
+
+    saveError (err: Error) {
+      return {
+        id: this.testId,
+        err: {
+          ...err,
+          message: saveErrorMessage(err.message),
+          docsUrl: 'https://on.cypress.io/studio-beta',
+        },
+      }
+    },
+
+    setFileDetails (fileDetails) {
```

Comment:
I'm not able to save the file as it fail with error:

```
ENOENT: no such file or directory, open '/Users/zachjw/work/component-test-repos/create-next-app/create-next-app/cypress/e2e/passing.cy.js'
```

Notice the projectName is being appended twice `create-next-app/create-next-app`. I traced it down to [stripCustomProtocol](https://github.com/cypress-io/cypress/blob/b0299c38009792df98ecd706dab29c04f24490dc/packages/driver/src/cypress/stack_utils.ts#L260), might need to tweak the regex

Reply:
Weird, I am **not** seeing this. I may need help to reproduce this one.